### PR TITLE
Hw security

### PR DESCRIPTION
**Pull request by GradientAspen**

### CODE REVIEW

File: src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        }
+
+        User user = new User();
+        user.setEmail(userRequestDto.getEmail());
+        user.setPassword(userRequestDto.getPassword());
+        user.setFirstName(userRequestDto.getFirstName());
+        user.setLastName(userRequestDto.getLastName());
+        user.setShippingAddress(userRequestDto.getShippingAddress());
+        User saveUser = userRepository.save(user);
+        return userMapper.toUserDto(saveUser);
```
Comment (olekskov) [inline src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java:31]: Why we use mapper for User->DTO and not using it for DTO->user?

File: src/main/java/com/project/springbootproject/dto/userdto/UserRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package com.project.springbootproject.dto.userdto;
+
+import com.project.springbootproject.validation.FieldMatch;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.hibernate.validator.constraints.Length;
+
+@Data
+
```
Comment (olekskov) [inline src/main/java/com/project/springbootproject/dto/userdto/UserRequestDto.java]:
```suggestion
```

File: src/main/java/com/project/springbootproject/exception/RegistrationException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.project.springbootproject.exception;
+
+public class RegistrationException extends Exception {
```
Comment (olekskov) [inline src/main/java/com/project/springbootproject/exception/RegistrationException.java]: 1. Why to make checked exception to escalate it whole way up to controller without handling?
Make it unchecked and avoid overdeclaration on methods/interfaces/etc
2. "Consider to handle the RegistrationException using the GlobalExceptionHandler"
cannot see this point of task

**Review by olekskov — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java (hunk 1 above)
Comment (GradientAspen) [reply src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java:31]: If I understand correctly we take userRequestDto in the method we go to User and then we mapped UserDto))

File: src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can not register user");
+        }
+
+        User user = new User();
+        user.setEmail(userRequestDto.getEmail());
+        user.setPassword(userRequestDto.getPassword());
+        user.setFirstName(userRequestDto.getFirstName());
+        user.setLastName(userRequestDto.getLastName());
+        user.setShippingAddress(userRequestDto.getShippingAddress());
```
Comment (olekskov) [inline src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java:29]: still not sure why you hesitate to use mapper
```suggestion
        User user = userMapper.toUserModel(userRequestDto);
```

**Review by olekskov — APPROVED** [review]

File: src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java (hunk 4 above)
Comment (GradientAspen) [reply src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java:29]: https://mate.academy/learn/java-spring-boot/java-spring-boot-security-basics?section=video&videoId=6179 
In this video from 9 min, Bohdan advises to use this approach, but he didnt explain it

**Pull request merged**